web/src/search/input/SearchButton.tsx

```diff
@@ -3,34 +3,54 @@ import HelpCircleOutlineIcon from 'mdi-react/HelpCircleOutlineIcon'
 import SearchIcon from 'mdi-react/SearchIcon'
 import * as React from 'react'
 import { Dropdown, DropdownItem, DropdownMenu, DropdownToggle } from 'reactstrap'
+import { ActivationStatus } from '../../../../shared/src/components/activation/Activation'
+import { ActivationClickTarget } from '../../../../shared/src/components/activation/ActivationClickTarget'
 
 interface Props {
     /** Hide the "help" icon and dropdown. */
     noHelp?: boolean
 
     /** Never show the "Search" button label. */
     noLabel?: boolean
+
+    activation?: ActivationStatus
 }
 
 interface State {
     isOpen: boolean
+
+    didClick: boolean
 }
 
 /**
  * A search button with a dropdown with related links.
  */
 export class SearchButton extends React.Component<Props, State> {
-    public state: State = { isOpen: false }
+    public state: State = { isOpen: false, didClick: false }
+
+    private button: React.RefObject<HTMLButtonElement>
```

Comment:
```suggestion
    private button = React.createRef<HTMLButtonElement>()
```

Prefer initialisers over constructors.